### PR TITLE
Improve code coverage

### PR DESCRIPTION
**Pull request by w3st3ry**
Signed-off-by: Valentin Pichard <7628998+w3st3ry@users.noreply.github.com>

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report test#145.0 ✘  
* Stage 1
  * Test ✘

File: api/api_test.go [hunk 1]
```diff
@@ -64,12 +64,24 @@ func TestMain(m *testing.M) {
...
 	hdl = srv.Handler(ctx)
 
 	os.Exit(m.Run())
 }
 
 func dumbIdentityProvider(r *http.Request) (string, error) {
+	username := r.Header.Get(usernameHeaderKey)
+
+	if username != adminUser && username != regularUser {
+		return "", errors.New("unknow user")
```
Comment (rbeuque74) [inline api/api_test.go]:
```suggestion
		return "", errors.New("unknown user")
```

Typo here

**Comment by ovh-cds** [comment]
CDS Report test#147.0 ✘  
* Stage 1
  * Test ✘

File: api/api_test.go (hunk 1 above)
Comment (w3st3ry) [reply api/api_test.go]: fixed

File: api/api_test.go [hunk 2]
```diff
@@ -99,9 +111,44 @@ func TestUtils(t *testing.T) {
...
+			iffy.ExpectStatus(301),
+		)
+
+	tester.AddCall("testMeta", http.MethodGet, "/meta", "").
+		Headers(regularHeaders).
+		Checkers(
+			iffy.ExpectStatus(200),
+		)
+
+	utask.FMaintenanceMode = true
```
Comment (nomonamo) [inline api/api_test.go]: be careful, this is misleading: the calls are declared with `tester.AddCall`, but they are only performed on `tester.Run()`
this means that `FMaintenanceMode` is effectively `true` for every call in this test
I would place this declaration either at the beginning of the test, with a deferred set to false
or right before `tester.Run()`, whatever seems more readable to you

Comment (w3st3ry) [reply api/api_test.go]: Fixed!
Thanks

**Pull request merged**